SlicerSofa/SlicerSofa.py

```diff
@@ -19,148 +51,490 @@
 
 from SofaEnvironment import *
 
-#
-# SlicerSofa
-#
+# -----------------------------------------------------------------------------
+# Decorator: RunOnce
+# -----------------------------------------------------------------------------
+def RunOnce(func):
+    """
+    Decorator that marks a function to be executed only once.
 
+    Attributes:
+        runOnce (bool): Flag indicating the function should run only once.
+    """
+    func.runOnce = True
+    return func
 
-class SlicerSofa(ScriptedLoadableModule):
-    """Uses ScriptedLoadableModule base class, available at:
-    https://github.com/Slicer/Slicer/blob/main/Base/Python/slicer/ScriptedLoadableModule.py
+# -----------------------------------------------------------------------------
+# Utility Functions
+# -----------------------------------------------------------------------------
+def arrayFromMarkupsROIPoints(roiNode):
```

Comment:
Maybe this should go in slicer.util.  Also shouldn't this be `arrayFromMarkupsROIBounds` since you can already get the points with `arrayFromMarkupsControlPoints`